tests/pytorch/distributed/run_layer_with_overlap.py

```diff
@@ -210,6 +267,12 @@ def _train(opts):
         opts.tcp_init = True
         opts.bind_to_device = True
         opts.bootstrap_backend = "mpi"
+    elif "SLURM_PROCID" in os.environ:
+        WORLD_RANK = (int)(os.environ["SLURM_PROCID"])
+        LOCAL_RANK = (int)(os.environ["SLURM_LOCALID"])
+        WORLD_SIZE = (int)(os.environ["SLURM_NTASKS"])
+        assert opts.tp is not None, "SLURM runs should provide --tp argument."
```

Comment:
If we make `--tp` a required arg, we need to set it in `test_comm_gemm_overlap.py`.